fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -227,8 +227,8 @@ public boolean isRefreshColumn(Map<String, Column> preNameToColumn,
                     needRefreshColumn = true;
                     break;
                 }
-                PrimitiveType type = convertColumnType(fieldSchema.getType());
-                if (type != column.getType().getPrimitiveType()) {
+                ScalarType type = convertColumnType(fieldSchema.getType());
+                if (type != column.getType()) {
```

Comment:
If this is ScalarType, `!=` means if they are the same object.
I think you want to call is `equals()`?

Reply:
How stupid am I.

Reply:
What we get is after hms has been normalized